integration-tests/src/test/scala/org/constellation/DownloadSpec.scala

```diff
@@ -0,0 +1,97 @@
+package org.constellation
+
+import cats.effect.{ContextShift, ExitCode, IO, Timer}
+import com.decodified.scalassh._
+import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+import org.constellation.schema.NodeState.Offline
+import org.constellation.schema.{ClusterNode, MetricsResult, NodeState, NodeStateInfo}
+import org.constellation.testutils.HttpUtil.{evalRequest, evalRequestForHosts}
+import org.constellation.testutils.{CustomMatchers, HttpUtil}
+import org.http4s.Uri
+import org.http4s.circe.CirceEntityDecoder._
+import org.http4s.client.dsl.io._
+import org.http4s.dsl.io._
+import org.scalatest.concurrent.PatienceConfiguration.Timeout
+import org.scalatest.concurrent.{Eventually, PatienceConfiguration}
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.time.SpanSugar._
+
+import scala.concurrent.ExecutionContext.global
+import scala.io.Source
+
+class DownloadSpec extends AnyFreeSpec with CustomMatchers with TestConfig with TerraformOutput with Eventually {
+
+  private val logger = Slf4jLogger.getLogger[IO]
+
+  override implicit val patienceConfig: PatienceConfig =
+    PatienceConfig(
+      timeout = scaled(3 minutes),
+      interval = scaled(1 second)
+    )
+
+  implicit val cs: ContextShift[IO] = IO.contextShift(global)
+  implicit val timer: Timer[IO] = IO.timer(global)
+
+  private val hostConfigProvider = HostConfigProvider
+    .fromHostConfig(
+      HostConfig(
+        login = PublicKeyLogin(config.ssh.user, config.ssh.keyfile),
+        hostKeyVerifier = HostKeyVerifiers.DontVerify
+      )
+    )
+
+  private val instanceIp: String = terraform.instanceIps.value.last
+
+  "restarting a node and joining cluster" in {
+    terraform.instanceIps.value.length should be > 3
+
+    val stopCmd = getResourceAsString("scripts/stop.sh")
+    evalSSHCommand(stopCmd)
+
+    eventually {
+      getClusterNodesState should contain only (Offline)
+    }
+
+    eventually(Timeout(1 hour)) {
```

Comment:
So this is the workaround to the problem I encountered and shared with @kpudlik . If we just wait for the node to be seen as offline by all of the nodes in the cluster, it may happen that it may still not be able to join the cluster, due to 401 that he's getting from the other nodes. There's some validation in `PeerAuthMiddleware` that may cause this to happen. It happens rarely (like 10% of time), but it does and it caused instability in tests.

Reply:
If so, 1h is too much. Anyway, if node is seen as offline by all other peers, it can make a rejoin. If it cannot rejoin even if all nodes seen the node as offline, it means there is a bug in the code, not in the tests.

Reply:
Yes, there's a bug in the code and I created a task for it: #1453 Until that bug is fixed we need to have stable tests. If you have any other ideas for a deterministic flow, please share.

Reply:
So, first of all, I'd add a TODO here, to remove it after #1453. I suppose there is a problem in these tests anyway, because the same flow works on mainnet. We should see nodes unable to join otherwise.

Reply:
Ok, I will add a TODO here.

Reply:
To summarize the offline discussion: we remove this check and fix the #1453 only if we see it fail again during the tests.